### PR TITLE
Engage batch

### PR DESCRIPTION
**Pull request by Arekusei**
This pull request based on #113 
Added batch for `/engage`

These changes will be checked for operability in the near future.
Is it normal if I simple copied tests `track_batch` and `track_batch_integration` to `test/people.js` and made small changes?

### CODE REVIEW

**Comment by tdumitrescu** [comment]
Thanks for the PR. Re: your question about tests, it would be preferable if you could DRY it out a bit by extracting the logic shared between your new tests and the existing ones (assuming it doesn't really harm legibility). Here's an example of a commit that pulls some copy-pasted test code into shared helpers: https://github.com/mixpanel/mixpanel-node/commit/2daa14b9640fdf267a67890fd3ab253127daa7d9 (though in your case, you'd need to extract to a separate module since it will be shared across different modules).

File: readme.md [hunk 1]
```diff
@@ -117,6 +117,39 @@ mixpanel.alias('distinct_id', 'your_alias');
...
+            properties: {
+                distinct_id: 'billybob',
+                color: 'red'
+            }
+        }
+    ]
+});
+
+// track multiple events at once
+mixpanel.people.batch({
```
Comment (cruzanmo) [inline readme.md]: We could name this method `mixpanel.people_batch` to be more consistent. What do others think?

Comment (Arekusei) [inline readme.md]: I think about it, but all the functions of people are in people object, and i don't know how will be better.

File: lib/mixpanel-node.js [hunk 2]
```diff
@@ -151,6 +175,121 @@ var create_client = function(token, config) {
...
+     * @param {Function} options.prepareItemFn
+     * @param {number}   [options.max_concurrent_requests]  limits concurrent async requests over the network
+     * @param {number}   [options.max_batch_size]           limits number of events sent to mixpanel per request
+     * @param {Function} [callback]                         callback receives array of errors if any
+     *
+     */
+    var send_batch_requests = function(options, callback) {
+        var list = options.list,
+            endpoint = options.endpoint,
+            prepareItemFn = options.prepareItemFn,
```
Comment (Arekusei) [inline lib/mixpanel-node.js]: It is a habit. I fix it now.

**Pull request closed without merging**